x/superfluid/keeper/msg_server.go

```diff
@@ -126,6 +126,11 @@ func (server msgServer) UnPoolWhitelistedPool(goCtx context.Context, msg *types.
 		return nil, err
 	}
 
+	err = server.keeper.checkUnpoolWhitelisted(ctx, msg.PoolId)
```

Comment:
Note: this is also checked down below in `UnpoolAllowedPools` but wouldn't hurt having it here too